### ISSUE TITLE
Adopt the “time window-based” policy for support of Python versions from NEP 29

### ISSUE THREAD

**Issue by mkoeppe** [sagemath/sage#30384]
<div id="comment:0"></div>

The NEP 29 policy (https://numpy.org/neps/nep-0029-deprecation_policy.html) aims to "standardiz[e] a recommendation for project support of minimum Python and NumPy versions [to] improve downstream project planning."
With this ticket, we adopt this deprecation policy of old Python versions for the Sage library.

According to NEP 29 support for Python 3.7 should be dropped in December 2021. Actually doing this will be subject of a new ticket that should be merged at the beginning of the 9.6 release cycle.

Related discussions:
- https://groups.google.com/d/msg/sage-devel/fgsSsJmVVXo/n2BQBSaBAAAJ (2020-05)
- https://groups.google.com/d/msg/sage-devel/aggikzrEfUg/_fIz4W-0BgAJ (2020-08)
- https://groups.google.com/g/sage-devel/c/j1cwbTU8aOU (2023-02)
- on PR #35403 (2023-03–)
- on PR #35404 (2023-03–)
- Vote: https://groups.google.com/g/sage-devel/c/3Zoq0CNE1hE (2023-05–26–2023-06-07)

CC:  @slel @jhpalmieri @orlitzky

Component: **python3**

Author: **Tobias Diez**

Branch/Commit: **[public/python3/nep29](https://github.com/sagemath/sagetrac-mirror/tree/public/python3/nep29) @ [`edf1711`](https://github.com/sagemath/sagetrac-mirror/commit/edf1711828659e3ac6914a9d9f021e9f38a27375)**

_Issue created by migration from https://trac.sagemath.org/ticket/30384_

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -3,3 +3,6 @@
 In this case we should indicate our intention to follow this policy  in our documentation.
 
 
+Related discussions:
+- https://groups.google.com/d/msg/sage-devel/fgsSsJmVVXo/n2BQBSaBAAAJ
+- https://groups.google.com/d/msg/sage-devel/aggikzrEfUg/_fIz4W-0BgAJ
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,4 +1,4 @@
-Getting the Python 3.8 support in #27754 merged in 9.2 will allow us to adopt the NEP 29 policy (https://numpy.org/neps/nep-0029-deprecation_policy.html)
+Getting the Python 3.8 support in #27754 merged in 9.2 will allow us to adopt the NEP 29 policy (https://numpy.org/neps/nep-0029-deprecation_policy.html) for the Sage library.
 
 In this case we should indicate our intention to follow this policy  in our documentation.
 
``````

**Comment by tobiasdiez** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,7 +1,7 @@
-Getting the Python 3.8 support in #27754 merged in 9.2 will allow us to adopt the NEP 29 policy (https://numpy.org/neps/nep-0029-deprecation_policy.html) for the Sage library.
+The NEP 29 policy (https://numpy.org/neps/nep-0029-deprecation_policy.html) aims to "standardiz[e] a recommendation for project support of minimum Python and NumPy versions [to] improve downstream project planning."
+With this ticket, we adopt this deprecation policy of old Python versions for the Sage library.
 
-In this case we should indicate our intention to follow this policy  in our documentation.
-
+According to NEP 29 support for Python 3.7 should be dropped in December 2021. Actually doing this will be subject of a new ticket that should be merged at the beginning of the 9.6 release cycle.
 
 Related discussions:
 - https://groups.google.com/d/msg/sage-devel/fgsSsJmVVXo/n2BQBSaBAAAJ
``````
